### PR TITLE
Deprecates our use of flask-executor (for the most part)

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Gets rid of flask-executor in all places, except for one occasion where we use it to calculate multiple metrics at once before the values are returned.  
Also adds some pretty major performance increases to the bulk disbursement flow. Subtle changes in terms of LoC, but major w.r.t performance (especially perceived performance to the end-user)

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoBlockchain/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by michielderoos** [comment]
[Refactor after request/async](https://trello.com/c/nALcksES/1071-refactor-after-request-async)

File: app/server/api/disbursement_api.py [hunk 1]
```diff
@@ -165,26 +162,13 @@ def post(self):
 class DisbursementAPI(MethodView):
     @requires_auth(allowed_roles={'ADMIN': 'admin'})
     def get(self, disbursement_id):
-        accounts = db.session.query(TransferAccount)\
```
Comment (michielderoos) [inline app/server/api/disbursement_api.py:168]: There's no reason to send accounts, and it's very slow. No FE uses this API, so it's safe to nix

File: app/server/models/disbursement.py [hunk 2]
```diff
@@ -46,7 +46,6 @@ class Disbursement(ModelBase, OneOrgBase):
     transfer_accounts = db.relationship(
         "TransferAccount",
         secondary=disbursement_transfer_account_association_table,
-        back_populates="disbursements",
```
Comment (michielderoos) [inline app/server/models/disbursement.py:49]: Deleting this one line made creating a new bulk disbursement (before approving-- the synchronous part) an order of magnitude faster.

File: app/server/utils/metrics/metrics.py [hunk 3]
```diff
@@ -129,14 +129,19 @@ def calculate_metric(metric):
...
     data = {}
-    for future in futures:
-        metric_name, result = future.result()
-        data[metric_name] = result
+    if g.get('is_after_request'):
+        for metric in metrics_list:
+            metric_name, result = calculate_metric(metric)
+            data[metric_name] = result
+    else:
+        futures = []
```
Comment (michielderoos) [inline app/server/utils/metrics/metrics.py:139]: This is the one spot we still want flask-executor. It's supremely janky after request, but quite stable in the right context. It really really helps with performance here.

File: app/server/api/disbursement_api.py [hunk 4]
```diff
@@ -77,7 +79,6 @@ def make_transfers(disbursement_id, auto_resolve=False):
         yield {
             'message': 'success' if percent_complete == 100 else 'pending',
             'percent_complete': math.floor(percent_complete),
-            'data': {'credit_transfers': credit_transfers_schema.dump(disbursement.credit_transfers).data}
```
Comment (michielderoos) [inline app/server/api/disbursement_api.py:80]: This is just read by the (as yet non-existent) progress bar. This data isn't needed there at all, and it's big and expensive. Deleting this was a big win.

File: app/test_app/functional/api/test_bulk_disbursement_api.py [hunk 5]
```diff
@@ -148,13 +141,16 @@ def test_disbursement_approval_flow(include_list, disbursement_amount, expected_
     """
     # Utility function to create admins to approve
     def make_new_user_and_token(name='francine', tier='superadmin'):
```
Comment (michielderoos) [inline app/test_app/functional/api/test_bulk_disbursement_api.py:143]: This part of the test is being janky as hell. I think it just has to do with how we handled sessions in the test special-case. The functionality works fine, it's just this little new user function. Working on a solution.

File: app/server/api/disbursement_api.py [hunk 6]
```diff
@@ -139,20 +140,16 @@ def post(self):
...
-            search_query = generate_search_query(search_string, filters, order, sort_by_arg, include_user=True)
+            search_query = generate_search_query(search_string, filters, order, sort_by_arg, include_user=False)
             search_query = search_query.filter(TransferAccount.id.notin_(exclude_accounts))
-            results = search_query.all()
-            transfer_accounts = [r[0] for r in results] # Get TransferAccount (TransferAccount, searchRank, User)
-        d.transfer_accounts.extend(transfer_accounts)
-
-        db.session.flush()
-
+            transfer_accounts = search_query.options(load_only("id")).all()
```
Comment (michielderoos) [inline app/server/api/disbursement_api.py:148]: No need to load the full-fat transfer accounts, all we need is IDs to use as foreign keys

**Comment by michielderoos** [comment]
[Refactor after request/async](https://trello.com/c/nALcksES/1071-refactor-after-request-async)

**Review by tristanhcole — APPROVED** [review]

**Pull request merged**